mock.go

```diff
@@ -625,6 +625,13 @@ func (q *MockFilter) Read(out interface{}) Op {
 			return err
 		}
 
+		// If a query-level clustering order has been provided, and the first one is descending, reverse the list
+		if len(m.options.ClusteringOrder) > 0 && m.options.ClusteringOrder[0].Direction == DESC {
+			for i, j := 0, len(result)-1; i < j; i, j = i+1, j-1 {
```

Comment:
we could bump the version of Go so that `slices.Reverse` is available, but I was worried that might have some knock on effects and make this a bigger change (due to dependencies and the CI etc. etc.)